pkg/controllers/status/work_status_controller.go

```diff
@@ -251,10 +251,27 @@ func (c *WorkStatusController) syncWorkStatus(key util.QueueKey) error {
 		// status changes.
 	}
 
+	if !c.isConsistentGenerationState(observedObj) {
```

Comment:
Obviously skipping state collection is not a better approach, even if the collected state is not reliable until a consistent state is reached, it is better than no state at all.
For users, it is obviously a more friendly way for Karmada to provide a continuous state change before reaching a consistent state.

Previous: 0 -> 1 -> 2 -> 3 -> 4
Now: 0 -> 4

Reply:
1.Skiping reflect status doesn't mean do not aggregate statuses.

2.Keeping the old status is better than aggregating untrustworthy statuses.

Reply:
> 1.Skiping reflect status doesn't mean do not aggregate statuses.

If status is not reflected, where does the aggregated data come from?

> Keeping the old status is better than aggregating untrustworthy statuses.

it's not untrustworthy status but the real status. And the old status is also untrustworthy.

Reply:
@whitewindmills  Perhaps this check condition can be deleted, but we should keep `status.observedGeneration < generation` for Karmada deployment if its member status is inconsistent.

Reply:
I have no objection to the modification of aggregating status.

Reply:
What about that reflect `.status.observedGeneration` to express the real status, and keep `status.observedGeneration < generation` condition by `AggregateStatus` operaton of deployment, not in the`work_status_controller.go`?

Reply:
I agree with @yike21
The `work status controller` just triggers status collection, but how status is collected is defined by resource interpreter. In addition, not all resources have the `.status.observedGeneration`, it's not appropriate to check it for all kinds of resources.